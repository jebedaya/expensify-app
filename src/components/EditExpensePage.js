import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpese, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpese(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.props.startRemoveExpense({id: this.props.expense.id});
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>  
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
                </div>
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpese: (id, expense) => dispatch(startEditExpese(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);