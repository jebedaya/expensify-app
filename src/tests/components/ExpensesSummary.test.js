import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseSummary with 2 expenses totalling 83.60 Shmekels', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={8360} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with 1 expense totalling 12.50 Shmekels', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1250} />);
    expect(wrapper).toMatchSnapshot();
});
