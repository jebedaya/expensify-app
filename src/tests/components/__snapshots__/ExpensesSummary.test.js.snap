// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense totalling 12.50 Shmekels 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling ₪
  12.50
</div>
`;

exports[`should render ExpenseSummary with 2 expenses totalling 83.60 Shmekels 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling ₪
  83.60
</div>
`;
